core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -484,10 +484,7 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
    * group does not know, because the information is not available yet or because the it has
    * failed to parse the Consumer Protocol, it returns true to be safe.
    */
-  def isSubscribedToTopic(topic: String): Boolean = subscribedTopics match {
-    case Some(topics) => topics.contains(topic)
-    case None => true
-  }
+  def isSubscribedToTopic(topic: String): Boolean = subscribedTopics.forall(topics => topics.contains(topic))
```

Comment:
How is this related to the PR description? Seems like an unrelated clean-up? Is your before/after test including these changes? Many of them _can_ result in additional allocations since you're replacing pattern matching with lambas.